upgrade_downgrade/test_dual_control_plane_upgrade_downgrade.sh

```diff
@@ -108,113 +109,52 @@ if [[ "${CLOUD}" == "GKE" ]];then
 
 # Install Istio 1.8 minimal profile with canary revision
-writeMsg "Deploy Istio(minimal) ${TO_TAG}"
+write_msg "Deploy Istio(minimal) ${TO_TAG}"
 ${TO_ISTIOCTL} install -y --set profile=minimal --set revision="${TO_REVISION}"
 kubectl wait --all --for=condition=Ready pods -n istio-system --timeout=5m
 
 # Relabel namespace before restarting each service
-writeMsg "Relabel namespace to inject ${TO_TAG} proxy"
+write_msg "Relabel namespace to inject ${TO_TAG} proxy"
 kubectl label namespace "${TEST_NAMESPACE}" istio-injection- istio.io/rev="${TO_REVISION}"
 
-function verifyIstiod() {
-  local ns="$1"
-  local app="$2"
-  local version="$3"
-  local istioctl_path="$4"
-  local expected="$5"
-
-  local mismatch=0
-
-  for pod in $(kubectl get pod -lapp="$app" -lversion="$version" -n "$ns" -o name); do
-    local istiod
-    local podname
-    podname=$(echo "$pod" | cut -d'/' -f2)
-    istiod=$(${istioctl_path} proxy-config bootstrap "$podname.$ns" | jq -r '.bootstrap.node.metadata.PROXY_CONFIG.discoveryAddress')
-    if [[ "$istiod" != *"$expected"* ]]; then
-      # Try once more. Peek into to pod spec and check
-      # if discovery address is set correctly. This is
-      # because of an error in 1.7 -> 1.8 test
-      line_count=$(kubectl get pod "$podname" -n "$ns" -o json | grep -c "discoveryAddress.*$expected")
-      if ((line_count == 0)); then
-        mismatch=$(( mismatch+1 ))
-      fi
-    fi
-  done
-
-  if ((mismatch == 0)); then
-    return 0
-  fi
-  return 1
-}
-
-restartDataPlane echosrv-deployment-v1 "${TEST_NAMESPACE}"
-withRetries 5 20 verifyIstiod "${TEST_NAMESPACE}" "echosrv-deployment-v1" "v1" \
-  "${TO_ISTIOCTL}" "istiod-${TO_REVISION}.istio-system.svc"
-withRetries 5 20 verifyIstiod "${TEST_NAMESPACE}" "echosrv-deployment-v2" "v2" \
-  "${TO_ISTIOCTL}" "istiod.istio-system.svc"
+restart_data_plane echosrv-deployment-v1 "${TEST_NAMESPACE}"
 
 if [[ "${TEST_SCENARIO}" == "dual-control-plane-upgrade" ]]; then
-  restartDataPlane echosrv-deployment-v2 "${TEST_NAMESPACE}"
-  withRetries 5 20 verifyIstiod "${TEST_NAMESPACE}" "echosrv-deployment-v2" "v2" \
-    "${TO_ISTIOCTL}" "istiod-${TO_REVISION}.istio-system.svc"
-  writeMsg "UPGRADE: Uninstall old version of control plane (${FROM_TAG})"
-  
-  # This test is for istio >= 1.6. So only 1.6 needs special handling
-  # else clause handles istio >= 1.7 which supports uninstall command
-  # (although still experimental)
+  restart_data_plane echosrv-deployment-v2 "${TEST_NAMESPACE}"
+  write_msg "UPGRADE: Uninstall old version of control plane (${FROM_TAG})"
   PROFILE_MANIFEST_YAML="${FROM_PATH}/manifests/profiles/minimal.yaml"
-  if [[ "${FROM_ISTIOCTL}" == *"1.6"* ]]; then
-    ${FROM_ISTIOCTL} manifest generate -f "${PROFILE_MANIFEST_YAML}" | kubectl delete -f -
-  else
-    ${FROM_ISTIOCTL} experimental uninstall -f "${PROFILE_MANIFEST_YAML}" -y --force
-  fi
+  ${FROM_ISTIOCTL} experimental uninstall -f "${PROFILE_MANIFEST_YAML}" -y --force
```

Comment:
add || true to continue for older versions.
also add the kubectl based delete above which will work for older versions.

Reply:
Done.